manager/controlapi/service.go

```diff
@@ -215,6 +215,10 @@ func (s *Server) checkPortConflicts(spec *api.ServiceSpec) error {
 	}
 
 	for _, service := range services {
+		// If service ID is the same (and not "") then this is an update
+		if serviceID != "" && serviceID == service.ID {
+			continue
```

Comment:
Should this be a `break` instead of a continue - just to avoid iterating the rest of the services for no reason?